minos/common/model/model.py

```diff
@@ -67,6 +72,31 @@ def __init__(self, *args, **kwargs):
         self._fields = dict()
         self._list_fields(*args, **kwargs)
 
+    @classmethod
+    def from_avro_bytes(cls, raw: bytes) -> t.Union["MinosModel", list["MinosModel"]]:
+        decoded = MinosAvroValuesDatabase().decode(raw, content_root=False)
+        if isinstance(decoded, list):
+            return [cls.from_dict(d) for d in decoded]
+        return cls.from_dict(decoded)
+
+    @classmethod
+    def from_dict(cls, d: dict[str, t.Any]) -> "MinosModel":
+        return cls(**d)
+
+    @classmethod
+    def to_avro_bytes(cls, models: list["MinosModel"]) -> bytes:
+        if len(models) == 0:
+            raise EmptyMinosModelSequenceException("'models' parameter cannot be empty.")
+
+        model_type = type(models[0])
+        if not all(model_type == type(model) for model in models):
+            raise MultiTypeMinosModelSequenceException(
+                f"Every model must have type {model_type} to be valid. Found types: {[type(model) for model in models]}"
+            )
+
+        avro_schema = models[0].avro_schema
+        return MinosAvroValuesDatabase().encode([model.avro_data for model in models], avro_schema)
+
```

Comment:
The initial approach to integrate `MinosModel` with the logic that serialises and deserialises the data to the `avro` format has been to perform a minimal set of modifications over `MinosAvroValuesDatabase` to be able to work fine with any `data` and `schema` structures.

Probably, a better approach is to perform a code refactor over the `minos.common.protocol` module, but I don't know if fits inside this issue, or a new one could be created specifically for that task.